src/script/interpreter.cpp

```diff
@@ -1338,7 +1338,29 @@ bool EvalScript(std::vector<std::vector<unsigned char> >& stack, const CScript&
                     stack.push_back(fSuccess ? vchTrue : vchFalse);
                     break;
                 }
+                
+                case OP_PAIRCOMMIT: {
+                    // OP_PAIRCOMMIT is only available in Tapscript
+                    if (sigversion == SigVersion::BASE || sigversion == SigVersion::WITNESS_V0) return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
```

Comment:
Doing the tapscript check here might lead to hardfork behavior 
[More info here](https://github.com/bitcoin-inquisition/bitcoin/pull/39#discussion_r1464005864)

Reply:
Isn't that only the case because of the disabling if block at the top for the disabled opcodes specifically? Shouldn't apply for opcodes that were never disabled IIUC.

Reply:
It would definitely be preferable to do LNhance without a hard fork! 😅

Reply:
We have this in the LNhance branch, so what happens is until it's active, it should fall back to OP_SUCCESS behavior. No?
```c++
static bool ExecuteWitnessScript(const Span<const valtype>& stack_span, const CScript& exec_script, unsigned int flags, SigVersion sigversion, const BaseSignatureChecker& checker, ScriptExecutionData& execdata, ScriptError* serror)
{
    std::vector<valtype> stack{stack_span.begin(), stack_span.end()};

    if (sigversion == SigVersion::TAPSCRIPT) {
        // OP_SUCCESSx processing overrides everything, including stack element size limits
        CScript::const_iterator pc = exec_script.begin();
        while (pc < exec_script.end()) {
            opcodetype opcode;
            if (!exec_script.GetOp(pc, opcode)) {
                // Note how this condition would not be reached if an unknown OP_SUCCESSx was found
                return set_error(serror, SCRIPT_ERR_BAD_OPCODE);
            }
            if ((flags & SCRIPT_VERIFY_LNHANCE) && (
                    opcode == OP_CHECKSIGFROMSTACK || opcode == OP_INTERNALKEY || opcode == OP_PAIRCOMMIT)) {
                continue;
            }
            // New opcodes will be listed here. May use a different sigversion to modify existing opcodes.
            if (IsOpSuccess(opcode)) {
                if (flags & SCRIPT_VERIFY_DISCOURAGE_OP_SUCCESS) {
                    return set_error(serror, SCRIPT_ERR_DISCOURAGE_OP_SUCCESS);
                }
                return set_success(serror);
            }
        }
```

Reply:
I will mark this as resolved, since nobody seems to have anything else to add.

Reply:
yes @reardencode is right. This was only a problem with OP_CODES that were being re-activated. Feel free to close